netebpfext/net_ebpf_ext.c

```diff
@@ -235,39 +235,63 @@ _Must_inspect_result_ ebpf_result_t
 net_ebpf_extension_wfp_filter_context_create(
     size_t filter_context_size,
     _In_ const net_ebpf_extension_hook_client_t* client_context,
+    _In_ const net_ebpf_extension_hook_provider_t* provider_context,
     _Outptr_ net_ebpf_extension_wfp_filter_context_t** filter_context)
 {
     ebpf_result_t result = EBPF_SUCCESS;
     net_ebpf_extension_wfp_filter_context_t* local_filter_context = NULL;
+    uint32_t client_context_count_max = NET_EBPF_EXT_MAX_CLIENTS_PER_HOOK_SINGLE_ATTACH;
 
     NET_EBPF_EXT_LOG_ENTRY();
 
     *filter_context = NULL;
 
+    if (net_ebpf_extension_hook_provider_get_attach_capability(provider_context) ==
+        ATTACH_CAPABILITY_MULTI_ATTACH_WITH_WILDCARD) {
+        client_context_count_max = NET_EBPF_EXT_MAX_CLIENTS_PER_HOOK_MULTI_ATTACH;
+    }
+
     // Allocate buffer for WFP filter context.
     local_filter_context = (net_ebpf_extension_wfp_filter_context_t*)ExAllocatePoolUninitialized(
         NonPagedPoolNx, filter_context_size, NET_EBPF_EXTENSION_POOL_TAG);
     NET_EBPF_EXT_BAIL_ON_ALLOC_FAILURE_RESULT(
         NET_EBPF_EXT_TRACELOG_KEYWORD_EXTENSION, local_filter_context, "local_filter_context", result);
 
     memset(local_filter_context, 0, filter_context_size);
+
+    local_filter_context->client_contexts = (net_ebpf_extension_hook_client_t**)ExAllocatePoolUninitialized(
+        NonPagedPoolNx,
+        client_context_count_max * sizeof(net_ebpf_extension_hook_client_t*),
+        NET_EBPF_EXTENSION_POOL_TAG);
+    NET_EBPF_EXT_BAIL_ON_ALLOC_FAILURE_RESULT(
+        NET_EBPF_EXT_TRACELOG_KEYWORD_EXTENSION,
+        local_filter_context->client_contexts,
+        "local_filter_context - client_contexts",
+        result);
+
```

Comment:
What happens if allocation fails?

Reply:
`NET_EBPF_EXT_BAIL_ON_ALLOC_FAILURE_RESULT` macro handles allocation failure and control goes to `Exit`